pkg/reconciler/apis/apibinding/apibinding_controller.go

```diff
@@ -129,10 +131,11 @@ func NewController(
 		deletedCRDTracker: newLockedStringSet(),
 	}
 
+	logger := logging.WithReconciler(klog.Background(), controllerName)
 	apiBindingInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    func(obj interface{}) { c.enqueueAPIBinding(obj, "") },
-		UpdateFunc: func(_, obj interface{}) { c.enqueueAPIBinding(obj, "") },
-		DeleteFunc: func(obj interface{}) { c.enqueueAPIBinding(obj, "") },
+		AddFunc:    func(obj interface{}) { c.enqueueAPIBinding(obj, logger, "") },
```

Comment:
@ncdc also open to ideas here, this seemed like the most natural extension of what you did, but unfortunately since the message was also appended to, it was not opaque to the caller

Reply:
why isn't logger part of the controller struct? Passing it around seems to be an anti-pattern.

Reply:
Nvmd. This reminds me of the stupid explicit contexts in Golang. No context around 🤷‍♂️